ocaml/testsuite/tests/typing-layouts/annots.ml

```diff
@@ -877,23 +877,13 @@ module type S = sig
 end
 ;;
 [%%expect {|
-Line 2, characters 10-36:
-2 |   val f : 'a. 'a t2_imm -> 'a t2_imm
-              ^^^^^^^^^^^^^^^^^^^^^^^^^^
-Error: The universal type variable 'a was defaulted to have kind value.
-       But it was inferred to have kind immediate
-         because of the definition of t2_imm at line 1, characters 0-28.
+module type S = sig val f : ('a : immediate). 'a t2_imm -> 'a t2_imm end
```

Comment:
The title of this test section above is now out of date.

Reply:
Note that we are still printing `('a : immediate). ...` even though `'a. ...` would be fine. I don't know if we do the same for other sources of defaulting -- e.g. type parameters -- but it would be better if we only added annotations when needed. If there is such support for type parameters then you should use it here too.